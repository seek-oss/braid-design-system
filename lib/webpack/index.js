const path = require('path');
const browserslist = require('browserslist-config-seek');
const isProduction = process.env.NODE_ENV === 'production';
const braidSrcPath = path.resolve(__dirname, '..');

const jsLoaders = [
  {
    loader: require.resolve('babel-loader'),
    options: {
      babelrc: false,
      presets: [
        [
          require.resolve('@babel/preset-env'),
          {
            modules: false,
            targets: browserslist
          }
        ],
        require.resolve('@babel/preset-react')
      ],
      plugins: [
        require.resolve('@babel/plugin-proposal-class-properties'),
        require.resolve('@babel/plugin-proposal-object-rest-spread')
      ]
    }
  }
];

const cssInJsLoaders = [
  {
    loader: require.resolve('style-loader'),
    options: {
      insertAt: 'top'
    }
  },
  {
    loader: require.resolve('css-loader'),
    options: {
      modules: true,
      localIdentName: `${
        isProduction ? '' : '__BRAID__[name]__[local]___'
      }[hash:base64:7]`,
      minimize: isProduction,
      importLoaders: 2 + jsLoaders.length
    }
  },
  {
    loader: require.resolve('postcss-loader'),
    options: {
      plugins: () => [require('autoprefixer')(browserslist)]
    }
  },
  {
    loader: require.resolve('css-in-js-loader')
  },
  ...jsLoaders
];

const decorateClientConfig = (config = {}) => {
  config.module = config.module || {};
  config.module.rules = config.module.rules || [];

  config.module.rules.push(
    {
      test: /(?!\.css)\.js$/,
      include: [braidSrcPath],
      use: jsLoaders
    },
    {
      test: /\.css\.js$/,
      include: [braidSrcPath],
      use: cssInJsLoaders
    }
  );

  return config;
};

module.exports = {
  decorateClientConfig
};
