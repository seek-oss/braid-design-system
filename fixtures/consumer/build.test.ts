/** @jest-environment node */
import path from 'path';
import glob from 'fast-glob';
import webpack from 'webpack';
import braidPkg from 'braid-design-system/package.json'; // eslint-disable-line no-restricted-imports

const { SideEffectsFlagPlugin } = webpack.optimize;

describe('build', () => {
  const cache = new Map();
  const checkSideEffects = (filePath: string): boolean => {
    // format the path the way Webpack does it internally
    const relativePath = `./${filePath}`;
    // https://github.com/webpack/webpack/blob/v5.72.1/lib/optimize/SideEffectsFlagPlugin.js#L89-L94
    const hasSideEffects = SideEffectsFlagPlugin.moduleHasSideEffects(
      relativePath,
      braidPkg.sideEffects,
      cache,
    );
    return hasSideEffects;
  };

  test('side-effects', async () => {
    const srcFiles = await glob(
      [
        '**/*.{js,ts,tsx}',
        '!**/*.{docs,gallery,playroom,screenshots,snippets,stories,test}.{ts,tsx}',
        '!node_modules',
      ],
      {
        cwd: path.dirname(require.resolve('braid-design-system/package.json')),
      },
    );

    const filesWithSideEffects = srcFiles
      .sort()
      .map((filePath) => checkSideEffects(filePath) && filePath)
      .filter(Boolean);

    expect(filesWithSideEffects).toMatchSnapshot();
  });
});
