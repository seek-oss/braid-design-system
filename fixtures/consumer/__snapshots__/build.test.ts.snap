// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build side-effects from dist 1`] = `
[
  "dist/playroom/components.cjs",
  "dist/playroom/components.mjs",
  "dist/reset.cjs",
  "dist/reset.mjs",
  "dist/side-effects/lib/components/BraidProvider/BraidProvider.cjs",
  "dist/side-effects/lib/components/BraidProvider/BraidProvider.mjs",
  "dist/side-effects/lib/css/reset/index.cjs",
  "dist/side-effects/lib/css/reset/index.mjs",
  "dist/side-effects/lib/css/reset/resetTracker.cjs",
  "dist/side-effects/lib/css/reset/resetTracker.mjs",
  "dist/side-effects/lib/themes/apac/index.cjs",
  "dist/side-effects/lib/themes/apac/index.mjs",
  "dist/side-effects/lib/themes/baseTokens/apac.cjs",
  "dist/side-effects/lib/themes/baseTokens/apac.mjs",
  "dist/side-effects/lib/themes/docs/index.cjs",
  "dist/side-effects/lib/themes/docs/index.mjs",
  "dist/side-effects/lib/themes/index.cjs",
  "dist/side-effects/lib/themes/index.mjs",
  "dist/side-effects/lib/themes/makeRuntimeTokens.cjs",
  "dist/side-effects/lib/themes/makeRuntimeTokens.mjs",
  "dist/side-effects/lib/themes/seekBusiness/index.cjs",
  "dist/side-effects/lib/themes/seekBusiness/index.mjs",
  "dist/side-effects/lib/themes/wireframe/index.cjs",
  "dist/side-effects/lib/themes/wireframe/index.mjs",
  "dist/styles/lib/components/Accordion/AccordionItem.css.cjs",
  "dist/styles/lib/components/Accordion/AccordionItem.css.mjs",
  "dist/styles/lib/components/Alert/Alert.css.cjs",
  "dist/styles/lib/components/Alert/Alert.css.mjs",
  "dist/styles/lib/components/Autosuggest/Autosuggest.css.cjs",
  "dist/styles/lib/components/Autosuggest/Autosuggest.css.mjs",
  "dist/styles/lib/components/Badge/Badge.css.cjs",
  "dist/styles/lib/components/Badge/Badge.css.mjs",
  "dist/styles/lib/components/Button/Button.css.cjs",
  "dist/styles/lib/components/Button/Button.css.mjs",
  "dist/styles/lib/components/ButtonIcon/ButtonIcon.css.cjs",
  "dist/styles/lib/components/ButtonIcon/ButtonIcon.css.mjs",
  "dist/styles/lib/components/Column/Column.css.cjs",
  "dist/styles/lib/components/Column/Column.css.mjs",
  "dist/styles/lib/components/ContentBlock/ContentBlock.css.cjs",
  "dist/styles/lib/components/ContentBlock/ContentBlock.css.mjs",
  "dist/styles/lib/components/Disclosure/Disclosure.css.cjs",
  "dist/styles/lib/components/Disclosure/Disclosure.css.mjs",
  "dist/styles/lib/components/Divider/Divider.css.cjs",
  "dist/styles/lib/components/Divider/Divider.css.mjs",
  "dist/styles/lib/components/Dropdown/Dropdown.css.cjs",
  "dist/styles/lib/components/Dropdown/Dropdown.css.mjs",
  "dist/styles/lib/components/Hidden/Hidden.css.cjs",
  "dist/styles/lib/components/Hidden/Hidden.css.mjs",
  "dist/styles/lib/components/HiddenVisually/HiddenVisually.css.cjs",
  "dist/styles/lib/components/HiddenVisually/HiddenVisually.css.mjs",
  "dist/styles/lib/components/List/List.css.cjs",
  "dist/styles/lib/components/List/List.css.mjs",
  "dist/styles/lib/components/Loader/Loader.css.cjs",
  "dist/styles/lib/components/Loader/Loader.css.mjs",
  "dist/styles/lib/components/MenuItem/useMenuItem.css.cjs",
  "dist/styles/lib/components/MenuItem/useMenuItem.css.mjs",
  "dist/styles/lib/components/MenuRenderer/MenuRenderer.css.cjs",
  "dist/styles/lib/components/MenuRenderer/MenuRenderer.css.mjs",
  "dist/styles/lib/components/MonthPicker/MonthPicker.css.cjs",
  "dist/styles/lib/components/MonthPicker/MonthPicker.css.mjs",
  "dist/styles/lib/components/OverflowMenu/OverflowMenu.css.cjs",
  "dist/styles/lib/components/OverflowMenu/OverflowMenu.css.mjs",
  "dist/styles/lib/components/Pagination/Pagination.css.cjs",
  "dist/styles/lib/components/Pagination/Pagination.css.mjs",
  "dist/styles/lib/components/Rating/Rating.css.cjs",
  "dist/styles/lib/components/Rating/Rating.css.mjs",
  "dist/styles/lib/components/Stepper/Stepper.css.cjs",
  "dist/styles/lib/components/Stepper/Stepper.css.mjs",
  "dist/styles/lib/components/Tabs/Tabs.css.cjs",
  "dist/styles/lib/components/Tabs/Tabs.css.mjs",
  "dist/styles/lib/components/Tag/Tag.css.cjs",
  "dist/styles/lib/components/Tag/Tag.css.mjs",
  "dist/styles/lib/components/TextDropdown/TextDropdown.css.cjs",
  "dist/styles/lib/components/TextDropdown/TextDropdown.css.mjs",
  "dist/styles/lib/components/TextLink/TextLink.css.cjs",
  "dist/styles/lib/components/TextLink/TextLink.css.mjs",
  "dist/styles/lib/components/Textarea/Highlight/Highlight.css.cjs",
  "dist/styles/lib/components/Textarea/Highlight/Highlight.css.mjs",
  "dist/styles/lib/components/Textarea/Textarea.css.cjs",
  "dist/styles/lib/components/Textarea/Textarea.css.mjs",
  "dist/styles/lib/components/Tiles/Tiles.css.cjs",
  "dist/styles/lib/components/Tiles/Tiles.css.mjs",
  "dist/styles/lib/components/Toggle/Toggle.css.cjs",
  "dist/styles/lib/components/Toggle/Toggle.css.mjs",
  "dist/styles/lib/components/TooltipRenderer/TooltipRenderer.css.cjs",
  "dist/styles/lib/components/TooltipRenderer/TooltipRenderer.css.mjs",
  "dist/styles/lib/components/icons/IconArrow/IconArrow.css.cjs",
  "dist/styles/lib/components/icons/IconArrow/IconArrow.css.mjs",
  "dist/styles/lib/components/icons/IconChevron/IconChevron.css.cjs",
  "dist/styles/lib/components/icons/IconChevron/IconChevron.css.mjs",
  "dist/styles/lib/components/icons/IconThumb/IconThumb.css.cjs",
  "dist/styles/lib/components/icons/IconThumb/IconThumb.css.mjs",
  "dist/styles/lib/components/private/Field/Field.css.cjs",
  "dist/styles/lib/components/private/Field/Field.css.mjs",
  "dist/styles/lib/components/private/InlineField/InlineField.css.cjs",
  "dist/styles/lib/components/private/InlineField/InlineField.css.mjs",
  "dist/styles/lib/components/private/Keyline/Keyline.css.cjs",
  "dist/styles/lib/components/private/Keyline/Keyline.css.mjs",
  "dist/styles/lib/components/private/Modal/Modal.css.cjs",
  "dist/styles/lib/components/private/Modal/Modal.css.mjs",
  "dist/styles/lib/components/private/Placeholder/Placeholder.css.cjs",
  "dist/styles/lib/components/private/Placeholder/Placeholder.css.mjs",
  "dist/styles/lib/components/private/Truncate/Truncate.css.cjs",
  "dist/styles/lib/components/private/Truncate/Truncate.css.mjs",
  "dist/styles/lib/components/private/hideFocusRings/hideFocusRings.css.cjs",
  "dist/styles/lib/components/private/hideFocusRings/hideFocusRings.css.mjs",
  "dist/styles/lib/components/private/touchable/virtualTouchable.css.cjs",
  "dist/styles/lib/components/private/touchable/virtualTouchable.css.mjs",
  "dist/styles/lib/components/useToast/Toast.css.cjs",
  "dist/styles/lib/components/useToast/Toast.css.mjs",
  "dist/styles/lib/css/atoms/atomicProperties.cjs",
  "dist/styles/lib/css/atoms/atomicProperties.mjs",
  "dist/styles/lib/css/atoms/atoms.cjs",
  "dist/styles/lib/css/atoms/atoms.mjs",
  "dist/styles/lib/css/atoms/sprinkles.css.cjs",
  "dist/styles/lib/css/atoms/sprinkles.css.mjs",
  "dist/styles/lib/css/lineHeightContainer.css.cjs",
  "dist/styles/lib/css/lineHeightContainer.css.mjs",
  "dist/styles/lib/css/negativeMargin/negativeMargin.css.cjs",
  "dist/styles/lib/css/negativeMargin/negativeMargin.css.mjs",
  "dist/styles/lib/css/reset/reset.css.cjs",
  "dist/styles/lib/css/reset/reset.css.mjs",
  "dist/styles/lib/css/textAlignedToIcon.css.cjs",
  "dist/styles/lib/css/textAlignedToIcon.css.mjs",
  "dist/styles/lib/css/typography.css.cjs",
  "dist/styles/lib/css/typography.css.mjs",
  "dist/styles/lib/hooks/useIcon/icon.css.cjs",
  "dist/styles/lib/hooks/useIcon/icon.css.mjs",
  "dist/styles/lib/themes/apac/apacTheme.css.cjs",
  "dist/styles/lib/themes/apac/apacTheme.css.mjs",
  "dist/styles/lib/themes/apac/tokens.cjs",
  "dist/styles/lib/themes/apac/tokens.mjs",
  "dist/styles/lib/themes/docs/docsTheme.css.cjs",
  "dist/styles/lib/themes/docs/docsTheme.css.mjs",
  "dist/styles/lib/themes/docs/tokens.cjs",
  "dist/styles/lib/themes/docs/tokens.mjs",
  "dist/styles/lib/themes/makeBraidTheme.cjs",
  "dist/styles/lib/themes/makeBraidTheme.mjs",
  "dist/styles/lib/themes/makeVanillaTheme.cjs",
  "dist/styles/lib/themes/makeVanillaTheme.mjs",
  "dist/styles/lib/themes/seekBusiness/seekBusinessTheme.css.cjs",
  "dist/styles/lib/themes/seekBusiness/seekBusinessTheme.css.mjs",
  "dist/styles/lib/themes/seekBusiness/tokens.cjs",
  "dist/styles/lib/themes/seekBusiness/tokens.mjs",
  "dist/styles/lib/themes/vars.css.cjs",
  "dist/styles/lib/themes/vars.css.mjs",
  "dist/styles/lib/themes/wireframe/tokens.cjs",
  "dist/styles/lib/themes/wireframe/tokens.mjs",
  "dist/styles/lib/themes/wireframe/wireframeTheme.css.cjs",
  "dist/styles/lib/themes/wireframe/wireframeTheme.css.mjs",
  "dist/themes/apac.cjs",
  "dist/themes/apac.mjs",
  "dist/themes/docs.cjs",
  "dist/themes/docs.mjs",
  "dist/themes/seekBusiness.cjs",
  "dist/themes/seekBusiness.mjs",
  "dist/themes/wireframe.cjs",
  "dist/themes/wireframe.mjs",
]
`;

exports[`build side-effects from src 1`] = `
[
  "src/entries/playroom/components.ts",
  "src/entries/reset.ts",
  "src/entries/themes/apac.ts",
  "src/entries/themes/docs.ts",
  "src/entries/themes/seekBusiness.ts",
  "src/entries/themes/wireframe.ts",
  "src/lib/components/Accordion/AccordionItem.css.ts",
  "src/lib/components/Alert/Alert.css.ts",
  "src/lib/components/Autosuggest/Autosuggest.css.ts",
  "src/lib/components/Badge/Badge.css.ts",
  "src/lib/components/BraidProvider/BraidProvider.tsx",
  "src/lib/components/Button/Button.css.ts",
  "src/lib/components/ButtonIcon/ButtonIcon.css.ts",
  "src/lib/components/Column/Column.css.ts",
  "src/lib/components/ContentBlock/ContentBlock.css.ts",
  "src/lib/components/Disclosure/Disclosure.css.ts",
  "src/lib/components/Divider/Divider.css.ts",
  "src/lib/components/Dropdown/Dropdown.css.ts",
  "src/lib/components/Hidden/Hidden.css.ts",
  "src/lib/components/HiddenVisually/HiddenVisually.css.ts",
  "src/lib/components/List/List.css.ts",
  "src/lib/components/Loader/Loader.css.ts",
  "src/lib/components/MenuItem/useMenuItem.css.ts",
  "src/lib/components/MenuRenderer/MenuRenderer.css.ts",
  "src/lib/components/MonthPicker/MonthPicker.css.ts",
  "src/lib/components/OverflowMenu/OverflowMenu.css.ts",
  "src/lib/components/Pagination/Pagination.css.ts",
  "src/lib/components/Rating/Rating.css.ts",
  "src/lib/components/Stepper/Stepper.css.ts",
  "src/lib/components/Tabs/Tabs.css.ts",
  "src/lib/components/Tag/Tag.css.ts",
  "src/lib/components/TextDropdown/TextDropdown.css.ts",
  "src/lib/components/TextLink/TextLink.css.ts",
  "src/lib/components/Textarea/Highlight/Highlight.css.ts",
  "src/lib/components/Textarea/Textarea.css.ts",
  "src/lib/components/Tiles/Tiles.css.ts",
  "src/lib/components/Toggle/Toggle.css.ts",
  "src/lib/components/TooltipRenderer/TooltipRenderer.css.ts",
  "src/lib/components/icons/IconArrow/IconArrow.css.ts",
  "src/lib/components/icons/IconChevron/IconChevron.css.ts",
  "src/lib/components/icons/IconThumb/IconThumb.css.ts",
  "src/lib/components/private/Field/Field.css.ts",
  "src/lib/components/private/InlineField/InlineField.css.ts",
  "src/lib/components/private/Keyline/Keyline.css.ts",
  "src/lib/components/private/Modal/Modal.css.ts",
  "src/lib/components/private/Placeholder/Placeholder.css.ts",
  "src/lib/components/private/Truncate/Truncate.css.ts",
  "src/lib/components/private/hideFocusRings/hideFocusRings.css.ts",
  "src/lib/components/private/touchable/virtualTouchable.css.ts",
  "src/lib/components/useToast/Toast.css.ts",
  "src/lib/css/atoms/atomicProperties.ts",
  "src/lib/css/atoms/atoms.ts",
  "src/lib/css/atoms/sprinkles.css.ts",
  "src/lib/css/lineHeightContainer.css.ts",
  "src/lib/css/negativeMargin/negativeMargin.css.ts",
  "src/lib/css/reset/index.ts",
  "src/lib/css/reset/reset.css.ts",
  "src/lib/css/reset/resetTracker.ts",
  "src/lib/css/textAlignedToIcon.css.ts",
  "src/lib/css/typography.css.ts",
  "src/lib/hooks/useIcon/icon.css.ts",
  "src/lib/themes/apac/apacTheme.css.ts",
  "src/lib/themes/apac/index.ts",
  "src/lib/themes/apac/tokens.ts",
  "src/lib/themes/baseTokens/apac.ts",
  "src/lib/themes/docs/docsTheme.css.ts",
  "src/lib/themes/docs/index.ts",
  "src/lib/themes/docs/tokens.ts",
  "src/lib/themes/index.ts",
  "src/lib/themes/makeBraidTheme.ts",
  "src/lib/themes/makeRuntimeTokens.ts",
  "src/lib/themes/makeVanillaTheme.ts",
  "src/lib/themes/seekBusiness/index.ts",
  "src/lib/themes/seekBusiness/seekBusinessTheme.css.ts",
  "src/lib/themes/seekBusiness/tokens.ts",
  "src/lib/themes/themeUtils.ts",
  "src/lib/themes/tokenType.ts",
  "src/lib/themes/vars.css.ts",
  "src/lib/themes/wireframe/index.ts",
  "src/lib/themes/wireframe/tokens.ts",
  "src/lib/themes/wireframe/wireframeTheme.css.ts",
]
`;
